src/solver/batchnorm/backward_ck.cpp

```diff
@@ -184,66 +166,404 @@ static ConvSolution MakeAnyInvokerFactory(const miopen::batchnorm::ProblemDescri
+
+    int i = 0;
+    for(; i < bn_fwd_ptrs.size(); i++)
+    {
+        if(bn_fwd_ptrs[i]->GetTypeString() == this->kernel_id)
+        {
+            break;
+        }
+    }
+    if(i == valid_kernels.size())
+    {
+        return false;
+    }
+    auto argument_ptr = bn_fwd_ptrs[i]->MakeArgumentPointer(args.lens,
+                                                            args.in_strides,
+                                                            args.in_strides,
+                                                            args.in_strides,
+                                                            args.reduceDims,
+                                                            args.arrScaleBiasMeanVarLengths,
+                                                            args.arrScaleBiasMeanVarStrides,
+                                                            args.arrScaleBiasMeanVarStrides,
+                                                            args.arrScaleBiasMeanVarStrides,
+                                                            nullptr,
+                                                            nullptr,
+                                                            nullptr,
+                                                            nullptr,
+                                                            nullptr,
+                                                            0.0,
+                                                            PassThroughOp{},
+                                                            nullptr,
+                                                            nullptr,
+                                                            nullptr);
+    return bn_fwd_ptrs[i]->IsSupportedArgument(argument_ptr.get());
+}
+
+void PerformanceConfigBnCKBwdBackward::HeuristicInit(
+    const miopen::batchnorm::ProblemDescription& problem_desc)
+{
+#if !MIOPEN_BACKEND_HIP || !MIOPEN_USE_COMPOSABLEKERNEL
+    std::ignore = problem_desc;
+#else
+    switch(problem_desc.GetXDesc().GetType())
+    {
+    case miopenHalf: Init<F16, F32, F32, F32, F16, F32, F32>(problem_desc); break;
+    case miopenBFloat16: Init<BF16, F32, F32, F32, BF16, F32, F32>(problem_desc); break;
+    case miopenFloat: Init<F32, F32, F32, F32, F32, F32, F32>(problem_desc); break;
+    case miopenDouble: Init<F64, F64, F64, F64, F64, F64, F64>(problem_desc); break;
+    case miopenFloat8:
+    case miopenBFloat8:
+    case miopenInt8:
+    case miopenInt32:
+    case miopenInt64:
+    default: MIOPEN_THROW("Unsupported datatype");
+    }
+
+#endif
+}
+
+bool PerformanceConfigBnCKBwdBackward::SetNextValue(
+    const miopen::batchnorm::ProblemDescription& problem_desc)
+{
+#if !MIOPEN_BACKEND_HIP || !MIOPEN_USE_COMPOSABLEKERNEL
+    std::ignore = problem_desc;
+    return false;
+#else
+    if(this->valid_kernels.empty())
+    {
+        this->HeuristicInit(problem_desc);
+        assert(!valid_kernels.empty());
+        return true;
```

Comment:
I would suggest either throwing or returning false when condition in the assert is false